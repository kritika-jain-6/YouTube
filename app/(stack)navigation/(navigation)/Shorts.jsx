import React, { Component} from "react";
import { StyleSheet, SafeAreaView } from "react-native";


class Shorts extends Component {
    

    render() {
        return (
            <SafeAreaView style={styles.container} testID="shorts-screen">
            </SafeAreaView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: "space-between",
        backgroundColor: "#fff",
    },
});

export default Shorts;
