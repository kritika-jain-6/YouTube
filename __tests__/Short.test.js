import React from 'react';
import { render } from '@testing-library/react-native';
import Shorts from '../app/(stack)navigation/(navigation)/Shorts.jsx';

describe('Shorts Component', () => {
  it('renders without crashing', () => {
    const { getByTestId } = render(<Shorts />);
    expect(getByTestId('shorts-screen')).toBeTruthy();
  });

  it('renders a SafeAreaView', () => {
    const { getByTestId } = render(<Shorts />);
    const safeArea = getByTestId('shorts-screen');
    expect(safeArea.type).toBe('RCTSafeAreaView');
  });

  it('applies correct styles to container', () => {
    const { getByTestId } = render(<Shorts />);
    const container = getByTestId('shorts-screen');
    
    expect(container.props.style).toEqual({
      flex: 1,
      justifyContent: "space-between",
      backgroundColor: "#fff",
    });
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<Shorts />);
    expect(toJSON()).toMatchSnapshot();
  });
});