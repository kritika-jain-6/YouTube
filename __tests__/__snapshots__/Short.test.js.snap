// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shorts Component matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
  testID="shorts-screen"
/>
`;
